art/experiment/Experiment.py

```diff
@@ -73,27 +73,43 @@ def check_if_must_be_run(self, step: "Step", checks: List[Check]):
             )
             return True
 
-        print(f"Step {step.name}_{step.get_step_id()} was already completed.")
+        # print(f"Step {step.name}_{step.get_step_id()} was already completed.")
         return False
 
     def run_all(self):
+        steps_status = {step["step"].get_full_step_name(): ("Not run", None) for step in self.steps}
         for step in self.steps:
             self.metric_calculator.compile(step["skipped_metrics"])
             step, checks = step["step"], step["checks"]
 
             self.state.current_step = step
 
             if not self.check_if_must_be_run(step, checks):
+                steps_status[step.get_full_step_name()] = ("Skipped", step.get_results())
```

Comment:
Use some @dataclasses for this as done in cheks.py. This will be much more readable then just tuple